### DISCUSSION TITLE
Build specific branches only without using Vercel CLI

### DISCUSSION THREAD

**Discussion by dragGH102** [vercel/community#3535] [Help] [answered]
### Summary

I am aware that Vercel auto-build on every git branch and that when using Vercel CLI is possible to deploy  [https://github.com/orgs/vercel/discussions/1185](only in certain scenarios).

My scenario is the following:
- Push to git -> Deploy to either a specific branch or PROD
- I have feature branches that consist of several commits and I don't want a build to be made for every commit (no build for the feature branch until the branch is completely ready for testing)

In Project settings there is no option to specify which branches to build (except generic "production" and "preview" + custom "ignored Build Step" and a bunch of other options for file changes). I am however not sure how I should set up the ignore build step for my scenario.

**Can I build only 2 specific branches, without CLI and with either 'ignored build step' or a vercel config file?**

### Steps to Reproduce

- Push to branch main and/or branch X => build
- Push to branch Y/Z => should not build (but it builds anyways)

**Comment by amyegan** [comment, marked as answer]
Hey @dragGH102. In your situation, I recommend using [Ignored Build Step](https://vercel.com/docs/concepts/projects/overview#ignored-build-step). You can use [System Environment Variables](https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables#) in a custom script to target the exact scenarios in which a build should occur. 

For example, let's say you have two deployment branches called `prod` and `staging`. If you wanted to ignore builds from all other branches, then you could something like
```
#!/bin/bash

echo "VERCEL_GIT_COMMIT_REF: $VERCEL_GIT_COMMIT_REF"

if [[ "$VERCEL_GIT_COMMIT_REF" == "staging" || "$VERCEL_GIT_COMMIT_REF" == "prod"  ]] ; then
  # Proceed with the build
    echo "✅ - Build can proceed"
  exit 1;

else
  # Don't build
  echo "🛑 - Build cancelled"
  exit 0;
fi
```

This guide can help get you started with a custom script for your project: https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel
